cmd/crane-agent/app/manager.go

```diff
@@ -0,0 +1,151 @@
+package app
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+
+	ensuaranceapi "github.com/gocrane/api/ensurance/v1alpha1"
+	ensuaranceset "github.com/gocrane/api/pkg/generated/clientset/versioned"
+	"github.com/gocrane/crane/cmd/crane-agent/app/options"
+	"github.com/gocrane/crane/pkg/ensurance/analyzer"
+	"github.com/gocrane/crane/pkg/ensurance/avoidance"
+	"github.com/gocrane/crane/pkg/ensurance/executor"
+	einformer "github.com/gocrane/crane/pkg/ensurance/informer"
+	"github.com/gocrane/crane/pkg/ensurance/manager"
+	"github.com/gocrane/crane/pkg/ensurance/statestore"
+	"github.com/gocrane/crane/pkg/utils/log"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(ensuaranceapi.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+// NewManagerCommand creates a *cobra.Command object with default parameters
+func NewManagerCommand(ctx context.Context) *cobra.Command {
+	opts := options.NewOptions()
+
+	cmd := &cobra.Command{
+		Use:  "crane-agent",
+		Long: `The crane agent is responsible agent in crane`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opts.Complete(); err != nil {
+				log.Logger().Error(err, "opts complete failed,exit")
+				os.Exit(255)
+			}
+			if err := opts.Validate(); err != nil {
+				log.Logger().Error(err, "opts validate failed,exit")
+				os.Exit(255)
+			}
+
+			if err := Run(ctx, opts); err != nil {
+				fmt.Fprintf(os.Stderr, "%v\n", err)
+				os.Exit(1)
+			}
+		},
+	}
+
+	cmd.Flags().AddGoFlagSet(flag.CommandLine)
+	opts.AddFlags(cmd.Flags())
+	return cmd
+}
+
+// Run runs the crane-agent with options. This should never exit.
+func Run(ctx context.Context, opts *options.Options) error {
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
```

Comment:
Please remove all controller-runtime related codes since we don't use it in agent